### ISSUE TITLE
Nightly test failure with Hip debug build: unit test hip.view_api_b

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#4772]
Nightly Hip debug build with rocm/4.5.0 + MI100 began failing in the `hip.view_api_b` unit test:

```
23:40:59 4: [ RUN      ] hip.view_api_b
23:40:59 4: /home/jenkins/caraway-new/workspace/Kokkos_develop_carawaymi100_rocm450_hipserial-dbg/kokkos/core/unit_test/TestViewAPI.hpp:391: Failure
23:40:59 4: Expected equality of these values:
23:40:59 4:   error_flag
23:40:59 4:     Which is: 3
23:40:59 4:   0
23:40:59 4: [  FAILED  ] hip.view_api_b (10 ms)
```

Impacting changes to the Hip backend for this round of testing came in from PR #4371 and #4761

**Reproducer**:
```
module load cmake/3.19.3 rocm/4.5.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Hip,Serial --arch=VEGA908 --compiler=hipcc --cxxflags="-g -Werror " --cxxstandard="14"  --kokkos-path=$KOKKOS_PATH --no-examples --with-options=enable_large_mem_tests --debug
```

**Comment by ndellingwood** [comment]
@crtrott @ajpowelsnl this should be investigated for 3.6.0

**Comment by Rombur** [comment]
I cannot reproduce this on MI100 but I do see a similar problem (different error_flag) on MI60. This is due to https://github.com/kokkos/kokkos/pull/4755 and reverting it fixes the problem.

**Comment by ndellingwood** [comment]
@Rombur the test passed our nightlies last night, so it looks like an intermittent failure on MI100 as well (at least in debug builds). Thanks for the update on changes triggering the failure, I didn't realize this was a spurious failure at the first occurrence due to an earlier PR

**Comment by ajpowelsnl** [comment]
> @Rombur the test passed our nightlies last night, so it looks like an intermittent failure on MI100 as well (at least in debug builds). Thanks for the update on changes triggering the failure, I didn't realize this was a spurious failure at the first occurrence due to an earlier PR

Morning @Rombur and @ndellingwood --Is this still a blocker, or has it moved into some sort of unstable state?

**Comment by Rombur** [comment]
We think we have a fix [here](https://github.com/kokkos/kokkos/pull/4773) but it has not been merged yet.

**Comment by ajpowelsnl** [comment]
Morning @ndellingwood -- Are we good to go with this fix?  If so, may I close issue?

**Comment by ndellingwood** [comment]
@ajpowelsnl yes, I have not seen the issue since the post (though it was apparently intermittent). If it returns I'll reopen. Thanks!
